.github/workflows/documentation.yml

```diff
@@ -1,110 +1,166 @@
+  # If it succed, an artifact is made with the generated documentation
+  # (html format only). This artifact is used in the deploying job
   ocaml_docs:
-    # Build the documentation from the ocaml comments
-    # If this test fails, no more test
     name: OCaml documentation
 
-    strategy:
-       matrix:
-        cache-name:
-          - odoc
-
     runs-on: ubuntu-latest
 
     steps:
+      # Checkout the code of the current branch
       - name: Checkout code
         uses: actions/checkout@v2
 
+      # Retrieve the opam cache with unique key
+      # A new cache is created/used if the `.opam` files changes or
+      # if we use another ocaml version
+      # This action only retrieve de .opam/ directory
       - name: Retrieve opam cache
         uses: actions/cache@v2
         id: cache-opam
         with:
           path: ~/.opam
-          key: v1-${{ runner.os }}-alt-ergo-${{ matrix.cache-name }}-${{ env.OCAML_DEFAULT_VERSION }}-${{ hashFiles('*.opam') }}
+          key: v1-${{ runner.os }}-alt-ergo-odoc-${{ env.OCAML_DEFAULT_VERSION }}-${{ hashFiles('*.opam') }}
 
+      # Get an OCaml environment with opam installed and the proper ocaml version
+      # opam will used opam cache environment if retrieved
       - name: Use OCaml ${{ env.OCAML_DEFAULT_VERSION }}
         uses: avsm/setup-ocaml@v1
         with:
           ocaml-version: ${{ env.OCAML_DEFAULT_VERSION }}
 
-      - name: Set git user
-        run: |
-          git config --global user.name github-actions
-          git config --global user.email github-actions-bot@users.noreply.github.com
-
-      - run: opam pin add . -y --no-action
-      - run: opam depext -y alt-ergo-lib alt-ergo-parsers alt-ergo altgr-ergo
-        # if: steps.cache-opam.outputs.cache-hit != 'true'
-      - run: opam install -y ./*.opam --deps-only --with-test
-        # if: steps.cache-opam.outputs.cache-hit != 'true'
+      # Add OPAMYES=true to the environment, this is usefill to replace `-y` option
+      # in any opam call
+      - name: Setup opam variables
+        run: export OPAMYES=true
+
+      # Pin the packages, this is needed for opam depext
+      # remove this step when opam 2.1 will be used
+      - name: opam pin no action
+        run: opam pin add . --no-action
+
+      # Install external dependencies if the cache is not retrieved
+      # remove this step when opam 2.1 will be used
+      - name: opam install depext
+        run: opam depext alt-ergo-lib alt-ergo-parsers alt-ergo altgr-ergo
+        if: steps.cache-opam.outputs.cache-hit != 'true'
+
+      # Install dependencies if the cache is not retrieved
+      - name: opam install deps
+        run: opam install ./*.opam --deps-only --with-test
+        if: steps.cache-opam.outputs.cache-hit != 'true'
+
+      # Install odoc if the cache is not retrieved
+      - name: opam install odoc
+        run: opam install odoc
+        if: steps.cache-opam.outputs.cache-hit != 'true'
```

Comment:
Odoc should become a `{with-doc}` dependency so that this step is no longer needed